src/main/java/io/jenkins/plugins/credentials/secretsmanager/factory/username_password/AwsUsernamePasswordCredentials.java

```diff
@@ -0,0 +1,45 @@
+package io.jenkins.plugins.credentials.secretsmanager.factory.username_password;
+
+import com.cloudbees.plugins.credentials.common.StandardUsernamePasswordCredentials;
+import com.cloudbees.plugins.credentials.impl.BaseStandardCredentials;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.util.Secret;
+import io.jenkins.plugins.credentials.secretsmanager.Messages;
+
+import javax.annotation.Nonnull;
+import java.util.function.Supplier;
+
+public class AwsUsernamePasswordCredentials extends BaseStandardCredentials implements StandardUsernamePasswordCredentials {
+
+    private final Supplier<Secret> password;
+    private final String username;
+
+    public AwsUsernamePasswordCredentials(String id, String description, Supplier<Secret> password, String username) {
+        super(id, description);
+        this.password = password;
+        this.username = username;
+    }
+
+    @NonNull
+    @Override
+    public Secret getPassword() {
+        return password.get();
```

Comment:
Your snapshot of this should call `password.get()` in the constructor I think.

Reply:
The idea is the CredentialSnapshotTaker classes are responsible for resolving the secret value. (CredentialsSnapshotTaker is the idiomatic place for this to happen in the Jenkins credentials API.) They then construct a new version of the credential with a constant-value supplie that just returns the value (no calls to AWS). This is then handed off to the agent.